mslib/mscolab/sockets_manager.py

```diff
@@ -130,36 +130,39 @@ def handle_message(self, _json):
         p_id = _json['p_id']
         reply_id = int(_json["reply_id"])
         user = User.verify_auth_token(_json['token'])
-        perm = self.permission_check_emit(user.id, int(p_id))
-        if perm:
-            new_message = self.cm.add_message(user, _json['message_text'], str(p_id), reply_id=reply_id)
-            new_message_dict = get_message_dict(new_message)
-            if reply_id == -1:
-                socketio.emit('chat-message-client', json.dumps(new_message_dict), room=str(p_id))
-            else:
-                socketio.emit('chat-message-reply-client', json.dumps(new_message_dict), room=str(p_id))
+        if user is not None:
+            perm = self.permission_check_emit(user.id, int(p_id))
+            if perm:
+                new_message = self.cm.add_message(user, _json['message_text'], str(p_id), reply_id=reply_id)
+                new_message_dict = get_message_dict(new_message)
+                if reply_id == -1:
+                    socketio.emit('chat-message-client', json.dumps(new_message_dict), room=str(p_id))
+                else:
+                    socketio.emit('chat-message-reply-client', json.dumps(new_message_dict), room=str(p_id))
 
     def handle_message_edit(self, socket_message):
         message_id = socket_message["message_id"]
         p_id = socket_message["p_id"]
         new_message_text = socket_message["new_message_text"]
         user = User.verify_auth_token(socket_message["token"])
-        perm = self.permission_check_emit(user.id, int(p_id))
-        if perm:
-            self.cm.edit_message(message_id, new_message_text)
-            socketio.emit('edit-message-client', json.dumps({
-                "message_id": message_id,
-                "new_message_text": new_message_text
-            }), room=str(p_id))
+        if user is not None:
+            perm = self.permission_check_emit(user.id, int(p_id))
+            if perm:
+                self.cm.edit_message(message_id, new_message_text)
+                socketio.emit('edit-message-client', json.dumps({
+                    "message_id": message_id,
+                    "new_message_text": new_message_text
+                }), room=str(p_id))
 
     def handle_message_delete(self, socket_message):
         message_id = socket_message["message_id"]
         p_id = socket_message["p_id"]
         user = User.verify_auth_token(socket_message['token'])
-        perm = self.permission_check_emit(user.id, int(p_id))
-        if perm:
-            self.cm.delete_message(message_id)
-            socketio.emit('delete-message-client', json.dumps({"message_id": message_id}), room=str(p_id))
+        if user is not None:
```

Comment:
I would probably simplify this to `if (use is not None) and self.permission_check_emit(user.id, int(p_id)):` due to the evaluation order of such expressions.

Reply:
No action necessary now.